config_test.go

```diff
@@ -48,18 +50,74 @@ func TestE2E(t *testing.T) {
 
 	// Typically we would share a base configuration but here we're pretending
 	// to be two different services.
-	serverConf := tlsconfig.Build(
+	serverConf, err := tlsconfig.Build(
 		tlsconfig.WithIdentity(serverTLSCrt),
 	).Server(
 		tlsconfig.WithClientAuthentication(pool),
 	)
+	if err != nil {
+		t.Fatalf("failed to build server config: %v", err)
+	}
 
-	clientConf := tlsconfig.Build(
+	clientConf, err := tlsconfig.Build(
 		tlsconfig.WithIdentity(clientTLSCrt),
 	).Client(
 		tlsconfig.WithAuthority(pool),
 	)
+	if err != nil {
+		t.Fatalf("failed to build client config: %v", err)
+	}
+
+	testClientServerTLSConnection(t, clientConf, serverConf)
+}
+
+func TestE2EFromFile(t *testing.T) {
+	t.Parallel()
+
+	ca, err := certtest.BuildCA("tlsconfig")
+	if err != nil {
+		t.Fatalf("failed to build CA: %v", err)
+	}
+	caFile, err := writeCAToTempFile(ca)
+	if err != nil {
+		t.Fatalf("failed to write CA file: %v", err)
+	}
+	defer os.Remove(caFile)
+
+	serverCertFile, serverKeyFile, err := generateKeypairToTempFilesFromCA(ca)
```

Comment:
make sure to clean-up these files after the test run, maybe all these functions could take a tmp dir prefix and then we can just delete that directory at the end of the test